### PR TITLE
code clean up

### PR DESCRIPTION
**Pull request by Powerade**

### CODE REVIEW

File: src/app/App.tsx [hunk 1]
```diff
@@ -1,11 +1,10 @@
 import * as React from 'react'
 import { BrowserRouter as Router, Route, Switch } from 'react-router-dom'
-import { bindActionCreators } from 'redux'
 import Toast from './common/toast/toast'
 const { connect } = require('react-redux')
+import 'rxjs'
```
Comment (Hyjaz) [inline src/app/App.tsx]: what are you using rxjs for in this file?

**Comment by ivangabriele** [comment]
Imo it's not a good idea to start giving another structure than the original ElectraJs (flatten the .wallet inside the global ElectraJsMiddleware), because it will stop matching exactly the documentation and will become a mess later on.

File: main/middlewares/communication.ts [hunk 2]
```diff
@@ -70,6 +71,10 @@ export default class Communication {
       { event: 'electraJs:wallet:lock', call: this.electraJs.wallet.lock.bind(this.electraJs.wallet) },
       { event: 'electraJs:wallet:unlock', call: this.electraJs.wallet.unlock.bind(this.electraJs.wallet) },
       { event: 'electraJs:wallet:getBalance', call: this.electraJs.wallet.getBalance.bind(this.electraJs.wallet)},
+      { event: 'electraJs:wallet:getInfo', call: this.electraJs.wallet.getInfo.bind(this.electraJs.wallet)},
+      // { event: 'electraJs:wallet:getTransaction', call: this.electraJs.wallet.getTransaction.bind(this.electraJs.wallet)},
```
Comment (Hyjaz) [inline main/middlewares/communication.ts]: You can remove it completely.

Comment (Hyjaz) [inline main/middlewares/communication.ts]: you can remove line 75 completely, the commented code.

File: src/app/App.tsx [hunk 3]
```diff
@@ -1,42 +1,53 @@
 import * as React from 'react'
 import { BrowserRouter as Router, Route, Switch } from 'react-router-dom'
-import { bindActionCreators } from 'redux'
 import Toast from './common/toast/toast'
 const { connect } = require('react-redux')
+import * as express from 'express'
+import 'rxjs'
+// const { ipcRenderer } = require('electron')
```
Comment (Hyjaz) [inline src/app/App.tsx]: Why do you have this code?  `// const { ipcRenderer } = require('electron')`

File: src/app/App.tsx [hunk 4]
```diff
@@ -1,42 +1,53 @@
...
+expressApp.listen(EXPRESS_PORT)
 
 /**
  * Point of entrance
  */
 
-@connect(mapStateToProps, mapDispatchToProps)
+// ipcRenderer.on('newTransaction', function (event: any, data: any): void {
+//   store.dispatch(getTransaction(data.msg))
+// })
```
Comment (Hyjaz) [inline src/app/App.tsx]: You can remove these comments.

File: src/app/payments/container.tsx [hunk 5]
```diff
@@ -1,37 +1,40 @@
...
     sendEca,
     setAmount,
     setMessageAndBadge,
     setToAddress
     // tslint:disable-next-line:align
   }, dispatch)
 
-@connect(mapStateToProps, mapDispatchToProps)
-export default class Payments extends React.Component<Props & DispatchProps, any> {
+class Payments extends React.Component<StateProps & DispatchProps, any> {
```
Comment (Hyjaz) [inline src/app/payments/container.tsx]: Remove the tslint:disable-next-line:typedef and tslint: disable-next-line:align.

mapDispatchToProps line aligned can be fixed if you replace it by:

```
const mapDispatchToProps: MapDispatchToProps<DispatchProps, {}> =
(dispatch: Dispatch<StateProps>): DispatchProps =>
  bindActionCreators
  (
    {
      clearSendCardFields,
      getAddresses,
      sendEca,
      setAmount,
      setMessageAndBadge,
      setToAddress
    },
    dispatch
  )
```

File: src/app/transactions/container.tsx [hunk 6]
```diff
@@ -1,25 +1,20 @@
...
   bindActionCreators({
-    getTransactions
-    // tslint:disable-next-line:align
-  }, dispatch)
+    getTransactions},dispatch)
 
-@connect(mapStateToProps, mapDispatchToProps)
-export default class Transactions extends React.Component<Props & DispatchProps, any> {
+class Transactions extends React.Component<StateProps & DispatchProps, any> {
   // tslint:disable-next-line:typedef
```
Comment (Hyjaz) [inline src/app/transactions/container.tsx]: You can remove this `// tslint:disable-next-line:typedef` and simply add the return type `void` to

`public componentDidMount(): void`

File: src/app/transactions/epics.ts [hunk 7]
```diff
@@ -34,3 +31,31 @@ export function getTransactions(action$: ActionsObservable<TransactionsActions |
...
+    .debounceTime(MAX_DELAY)
+    .map((action: any) => ({
+      payload: action.payload
+    }))
+    .map(async (data: any) => ElectraJsMiddleware.getTransactions(100))
+    .switchMap((promise: any) =>
+      Observable
+        .fromPromise(promise)
+        .map((data: any) => {
+          // tslint:disable-next-line
```
Comment (Hyjaz) [inline src/app/transactions/epics.ts]: Remove // tslint:disable-next-line

File: src/app/transactions/epics.ts [hunk 8]
```diff
@@ -34,3 +31,31 @@ export function getTransactions(action$: ActionsObservable<TransactionsActions |
...
+          new Notification('New Transaction', { body: `Incoming transaction of ${data.amount} ECA` })
+
+          return { type: TransactionActionNames.GET_TRANSACTIONS }
+        })
+        .catch((error: Error) => {
+          console.log(error.message)
```
Comment (Hyjaz) [inline src/app/transactions/epics.ts]: Convert `console.log(error.message)` to `console.error(error.message)`

File: src/app/transactions/epics.ts [hunk 9]
```diff
@@ -34,3 +31,31 @@ export function getTransactions(action$: ActionsObservable<TransactionsActions |
...
+
+          return Observable.of({
+            type: TransactionActionNames.GET_TRANSACTIONS_FAIL
+          })
+        })
+    )
+}
```
Comment (Hyjaz) [inline src/app/transactions/epics.ts]: Missing LF

**Pull request closed without merging**